tests/foreman/ui_airgun/test_host.py

```diff
@@ -1297,3 +1550,111 @@ def test_positive_bulk_delete_host(session, module_loc):
         session.host.apply_action('Delete Hosts', list(hosts_names))
         values = session.host.read_all()
         assert not values['table']
+
+
+@tier3
+def test_positive_provision_end_to_end(
+        session,
+        module_org,
+        module_loc,
+        module_libvirt_domain,
+        module_libvirt_hostgroup,
+        module_libvirt_resource,
+
+):
+    """Provision Host on libvirt compute resource
+
+    :id: 2678f95f-0c0e-4b46-a3c1-3f9a954d3bde
+
+    :expectedresults: Host is provisioned successfully
+
+    :CaseLevel: System
+    """
+    hostname = gen_string('alpha').lower()
+    root_pwd = gen_string('alpha', 15)
+    puppet_env = entities.Environment(
+        location=[module_loc], organization=[module_org]).create()
+    with session:
+        session.host.create({
+            'host.name': hostname,
+            'host.organization': module_org.name,
+            'host.location': module_loc.name,
+            'host.hostgroup': module_libvirt_hostgroup.name,
+            'host.inherit_deploy_option': False,
+            'host.deploy': module_libvirt_resource,
+            'host.inherit_puppet_environment': False,
+            'host.puppet_environment': puppet_env.name,
+            'virtual_machine.memory': '1 GB',
+            'operating_system.root_password': root_pwd,
+            'interfaces.interface.network_type': 'Physical (Bridge)',
+            'interfaces.interface.network': settings.vlan_networking.bridge,
+        })
+        name = u'{0}.{1}'.format(hostname, module_libvirt_domain.name)
+        assert session.host.search(name)[0]['Name'] == name
+        try:
+            for _ in range(25):
+                values = session.host.get_details(name)
+                if values['properties']['properties_table']['Build'] == 'Pending installation':
+                    sleep(30)
```

Comment:
Just a query ...why to use `sleep` , can we not use `wait_for` here ?

Reply:
I believe we can try to use `wait_for`, so will look to change the code when gather all the comments, because I need full infrastructure to make even one re-test